src/physics/linear_filament.rs

```diff
@@ -138,3 +145,504 @@ pub fn inductance_piecewise_linear_filaments(
+            &xfil2[..=NFIL - 2],
+            &yfil2[..=NFIL - 2],
+            &zfil2[..=NFIL - 2],
+        );
+
+        let dlxfil2: Vec<f64> = (0..=NFIL - 2).map(|i| xfil2[i + 1] - xfil2[i]).collect();
+        let dlyfil2: Vec<f64> = (0..=NFIL - 2).map(|i| yfil2[i + 1] - yfil2[i]).collect();
+        let dlzfil2: Vec<f64> = (0..=NFIL - 2).map(|i| zfil2[i + 1] - zfil2[i]).collect();
+        let dlxyzfil2 = (&dlxfil2[..], &dlyfil2[..], &dlzfil2[..]);
+
+        let xmid2: Vec<f64> = xfil2
+            .iter()
+            .zip(dlxfil2.iter())
+            .map(|(x, dx)| x + dx / 2.0)
+            .collect();
+        let ymid2: Vec<f64> = yfil2
+            .iter()
+            .zip(dlyfil2.iter())
+            .map(|(x, dx)| x + dx / 2.0)
+            .collect();
+        let zmid2: Vec<f64> = zfil2
+            .iter()
+            .zip(dlzfil2.iter())
+            .map(|(x, dx)| x + dx / 2.0)
+            .collect();
+
+        // Check against Neumann's formula for mutual inductance
+        let outx = &mut [0.0; NFIL - 1];
+        let outy = &mut [0.0; NFIL - 1];
+        let outz = &mut [0.0; NFIL - 1];
+        vector_potential_linear_filament(
+            (&xmid2, &ymid2, &zmid2),
+            (&xyz, &xyz, &xyz),
+            (&dlxyz, &dlxyz, &dlxyz),
+            &[1.0],
+            (outx, outy, outz),
+        )
+        .unwrap();
+        let a_dot_dl: Vec<f64> = (0..NFIL - 1)
+            .map(|i| outx[i] * dlxfil2[i] + outy[i] * dlyfil2[i] + outz[i] * dlzfil2[i])
+            .collect();
+        let m_from_a = a_dot_dl.iter().sum();
+        let m = inductance_piecewise_linear_filaments(
+            (&xyz, &xyz, &xyz),
+            (&dlxyz, &dlxyz, &dlxyz),
+            xyzfil2,
+            dlxyzfil2,
+            false,
+        )
+        .unwrap();
+        assert!(approx(m, m_from_a, 1e-10, 1e-15));
+
+        let vp = |x: f64, y: f64, z: f64| {
+            let mut outx = [0.0];
+            let mut outy = [0.0];
+            let mut outz = [0.0];
+
+            // Vector potential, either serial or parallel variant
+            vector_potential_linear_filament(
+                (&[x], &[y], &[z]),
+                (&xyz, &xyz, &xyz),
+                (&dlxyz, &dlxyz, &dlxyz),
+                &[1.0],
+                (&mut outx, &mut outy, &mut outz),
+            )
+            .unwrap();
+
+            (outx[0], outy[0], outz[0])
+        };
+
+        let vals = [
+            0.25, 0.5, 2.5, 10.0, 100.0, 1000.0, -1000.0, -100.0, -10.0, -2.5, -0.5, -0.25,
+        ];
+        // finite diff delta needs to be small enough to be accurate
+        // but large enough that we can tell the difference between adjacent points
+        // that are very far from the origin
+        let eps = 1e-7;
+        for x in vals.iter() {
+            for y in vals.iter() {
+                for z in vals.iter() {
+                    let y = &(y + 1e-2); // Slightly adjust to avoid nans
+                    let x = &(x + 1e-2);
```

Comment:
These lines would be easier to read if they kept the x, y, z order.
```suggestion
                    let x = &(x + 1e-2); // Slightly adjust to avoid nans
                    let y = &(y + 1e-2);
```